src/heartbeat.ts

```diff
@@ -147,7 +180,9 @@ export class HeartbeatManager extends Disposable {
             e.message = `Failed to send ${suffix}, triggered by event: ${this.lastActivityEvent}: ${originMsg}`;
             this.logService.error(e);
             e.message = `Failed to send ${suffix}: ${originMsg}`;
-            this.telemetryService.sendTelemetryException(e, { workspaceId: this.connectionInfo.workspaceId, instanceId: this.connectionInfo.instanceId});
+            this.telemetryService.sendTelemetryException(e, { workspaceId: this.connectionInfo.workspaceId, instanceId: this.connectionInfo.instanceId });
+        } finally {
+            this.ideHeartbeatData.totalCount++;
```

Comment:
I think both variables should be increased here in one tick.

Reply:
I mean they should be increased in the same JS event loop tick, since successfulCount is increased within `await withServerApi` it does not guarantee

Reply:
Addressed, going to merge this PR